### DISCUSSION TITLE
Evaluate expressions in with-configuration

### DISCUSSION THREAD

**Discussion by elgohr** [community/community#25598] [Actions]
<p>Is it possible to evualuate environment variables in the <code>with</code>-configuration (see <a href="https://github.com/elgohr/Publish-Docker-Github-Action/issues/43#issuecomment-552186587" rel="nofollow noopener">https://github.com/elgohr/Publish-Docker-Github-Action/issues/43#issuecomment-552186587</a>)?<br>
The action itself could do this aswell, but this would make every actions input much more dynamic, as user scripts could result in every datatype. Would be great if we could use <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/metadata-syntax-for-github-actions#inputs" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/metadata-syntax-for-github-actions#inputs</a> to be sure to get a specific datatype as an input.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>To evaluate expressions in with configuration, you need to put your expression inside {{ }} .</p>
<p>Please try this:</p>
<pre><code>name: CI
on:
  push:
    branches:
      - master

  schedule:
    - cron: '*/5 * * * *'

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - uses: actions/hello-world-docker-action@master
      with:
        who-to-greet: ${{ github.event_name != 'schedule' }}
</code></pre>
<p>You can refer to “About context and expressions” document:</p>
<p><a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fcontexts-and-expression-syntax-for-github-actions%23about-context-and-expressions&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7C1a89d9a2c3784fef0c9608d766840c1c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637090589630814951&amp;sdata=EAA44KpSltwSSBV0auwwULubh2lFu%2FK8dASm64zQfpQ%3D&amp;reserved=0" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#about-context-and-expressions</a></p>
